### ISSUE TITLE
Can not increase partitions in topic with only one partition

### ISSUE THREAD

**Issue by poulhenriksen** [apache/pulsar#6564]
**Describe the bug**
A partitioned topic with only one partition can not be increased to have more partitions.

**To Reproduce**
1. Create a partitioned topic with one partition:
    ```
    bin/pulsar-admin persistent create-partitioned-topic \
    persistent://public/default/part_topic \
    --partitions 1
    ```
2. Try to increase the number of partitions:
    ```
    bin/pulsar-admin topics update-partitioned-topic persistent://public/default/part_topic -p 3
    ```

It then fails with:

```
13:57:59.057 [AsyncHttpClient-5-1] WARN  org.apache.pulsar.client.admin.internal.BaseResource - [http://localhost:8080/admin/v2/persistent/public/default/part_topic/partitions?updateLocalTopicOnly=false] Failed to perform http post request: javax.ws.rs.ClientErrorException: HTTP 409 Conflict
Topic is not partitioned topic

Reason: Topic is not partitioned topic
```

**Expected behavior**
It does not fail.

**Comment by sijie** [comment]
This has been fixed by #6560

**Comment by zhanghaou** [comment]
It seems that this bug can be reproduced. I'm interesting in fix it.
